cloudify_cli/commands/apply.py

```diff
@@ -0,0 +1,256 @@
+          workflow_id,
+          skip_install,
+          skip_uninstall,
+          dont_skip_reinstall,
+          ignore_failure,
+          install_first,
+          preview,
+          dont_update_plugins,
+          force,
+          tenant_name,
+          visibility,
+          validate,
+          include_logs,
+          json_output,
+          runtime_only_evaluation,
+          auto_correct_types,
+          reevaluate_active_statuses,
+          skip_plugins_validation,
+          parameters,
+          allow_custom_parameters,
+          blueprint_labels,
+          deployment_labels,
+          logger,
+          client
+          ):
+    """Apply command uses the `cfy install` or `cfy deployments update`
+    depending on the existence the deployment specified by `DEPLOYMENT_ID`.
+
+    If the deployment exists, the deployment will be updated with the
+    given blueprint.
+    Otherwise, the blueprint will be installed (the deployment name will be
+    DEPLOYMENT_ID).
+    In both cases the blueprint is being uploaded to the manager.
+
+    `BLUEPRINT_PATH` can be a:
+        - local blueprint yaml file
+        - blueprint archive
+        - url to a blueprint archive
+        - github repo (`organization/blueprint_repo[:tag/branch]`)
+
+    Supported archive types are: zip, tar, tar.gz and tar.bz2
+
+    `DEPLOYMENT_ID` is the deployment's id to install/update.
+
+    Default values:
+
+    If BLUEPRINT_PATH not provided, the default blueprint path is
+    'blueprint.yaml' in the current work directory.
+    If DEPLOYMENT_ID is not provided it will be inferred from BLUEPRINT_PATH
+    as follows:
+    - Directory name for local blueprint path or archive with default
+      --blueprint-filename(blueprint.yaml).
+    - <Directory name>.<blueprint_filename> if it's an archive and
+      --blueprint-filename is not default.
+    """
+    if not blueprint_path:
+        blueprint_path = blueprint_path or os.path.join(os.getcwd(),
+                                                        DEFAULT_BLUEPRINT_PATH)
+        logger.info("No blueprint path provided, using default: %s",
+                    blueprint_path)
+
+    with process_blueprint_and_infer_deployment_id(blueprint_path,
+                                                   blueprint_filename,
+                                                   blueprint_id,
+                                                   deployment_id) as \
+            processed_inputs:
+        logger.debug("processed_inputs %s", processed_inputs)
+        try:
+            logger.info("Trying to find deployment %s",
+                        processed_inputs['deployment_id'])
+            deployment = client.deployments.get(
+                deployment_id=processed_inputs['deployment_id'])
+        except CloudifyClientError as e:
+            if e.status_code != 404:
+                raise CloudifyCliError(
+                    'deployment %s not found and error code is not 404',
```

Comment:
The deployment might exist, but an error was thrown.

Reply:
Done